### ISSUE TITLE
Cannot deserialize a type with interface member and JsonConverter annotation (used to work in 5.6.0) 

### ISSUE THREAD

**Issue by stefando** [elastic/elasticsearch-net#3094]
**NEST/Elasticsearch.Net version**: 6.0.1

**Elasticsearch version**: 5.6.4

**Description of the problem including expected versus actual behavior**:

We have the the class with the members of interface type annotated with the appropriate ```JsonConverter``` 

```c#
public class SomeClass {
        [JsonProperty(PropertyName = "points")]
        [JsonConverter(typeof(ArrayTypeConverter<IHouseNumberPoint,HouseNumberPoint>))]
        public IHouseNumberPoint[] Points { get; set; }
}
```
The classes used to deserialize ok in 5.6.0.
In 6.0.x there is Nest.Json.JsonSerializationException.

```
Nest.Json.JsonSerializationException: 
Could not create an instance of type Sygic.Locations.Elasticsearch.Shared.IHouseNumberPoint. 
Type is an interface or abstract class and cannot be instantiated. 
Path 'points[0].location', line 11, position 17. 
```

Is there a documentation/guidelines how to use JsonConverters in NEST 6+?

**Comment by russcam** [comment]
Take a look at the [documentation on Custom Serialization](https://www.elastic.co/guide/en/elasticsearch/client/net-api/current/custom-serialization.html).

Specifically, you'll want to use `JsonNetSerializer` from the [`NEST.JsonNetSerializer` nuget package](https://www.nuget.org/packages/NEST.JsonNetSerializer/) and hook it up as shown in the documentation to delegate serialization of your types to Json.NET. Please let us know if the documentation is unclear or how it might be improved.

**Comment by russcam** [comment]
@stefando just circling back to this. Did you manage to get this working?

**Comment by stefando** [comment]
Hi @russcam .
Yes, it is working for us (except for the issue #3096).

**Comment by russcam** [comment]
Thanks @stefando. We are aiming to release new versions of 5.x and 6.x tomorrow, including a fix for #3096. 

Just noticed that you're using NEST 6.0.1 against 5.6.4; major versions of the client are compatible and supported with major versions of Elasticsearch, so I would recommend using NEST 5.6.1 with Elasticsearch 5.6.4
